arch/arm/core/aarch32/mpu/arm_core_mpu.c

```diff
@@ -304,8 +306,14 @@ void z_arm_configure_dynamic_mpu_regions(struct k_thread *thread)
 #endif /* CONFIG_MPU_STACK_GUARD */
 
 	/* Configure the dynamic MPU regions */
+#ifdef CONFIG_AARCH32_ARMV8_R
```

Comment:
While we are reconfiguring the dynamic regions we can get MPU fault there. I'm surprised that it doesn't happen with ARMv8-M. Does I'm missing something ?

This behaviour is almost the same than for AArch64 Cortex-R support, where we reconfigure dynamic region with mpu disabled.

Reply:
IMHO, the MPU fault is generated because the `arm_core_mpu_configure_dynamic_mpu_regions` uses the data which is under an MPU region[1] to reconfigure the MPU regions[2]. If [2] includes [1], then it easily gets an MPU fault. When you reconfigure a region, you need 1) disable the region, 2) re-config, and 3) enable it. When the 2) re-config code accesses the data under the region we just disabled, v8r will gen an MPU fault.

I know the v8-M doesn't, but I don't know the reason. Maybe it depends on the implementation that MPU won't be activated under privilege mode.

Reply:
Yes that was exactly my analysis too. Thanks.